msal/application.py

```diff
@@ -549,6 +544,9 @@ def __init__(
                     )
             else:
                 raise
+
+        if allow_broker and sys.platform != "win32":
+            raise ValueError("allow_broker=True is only supported on Windows")
```

Comment:
Personally, I think this check defeats the purpose of the word "allow". If an error is raised, it should be called `use_broker` instead.

That being said, I prefer `use_broker` check more as it is more explicit (explicit is better than implicit).

Reply:
That is indeed a naming dilemma, unfortunately.

1. The word "allow" was influenced by the less-tangible fact that MsalRuntime does not and may never support some scenarios, such as ADFS, B2C, device code flow, SSH cert (work-in-progress), etc.. MSAL automatically falls back to browser in those scenarios, so that the app developer does not need to.
2. But we recently identified a situation that MSAL should not fallback to browser when running on a not-yet-supported OS/platform. Please refer to section 3.2 in this [internal design doc](https://identitydivision.visualstudio.com/DevEx/_git/AuthLibrariesApiReview/pullrequest/6882?_a=files).

Even with the newly found No.2, the No.1 still applies. So, there is no good name here. Ideas are welcome, although we may not actually change it at this point, because it has been shipped in MSAL Python 1.20.